File: java/event_id.java
/* ============================================================
 *
 * This file is part of the RSB project.
 *
 * Copyright (C) 2011 Jan Moringen <jmoringe@techfak.uni-bielefeld.de>
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU General Public License as
 * published by the Free Software Foundation; either version 2, or (at
 * your option) any later version.
 *
 * This program is distributed in the hope that it will be useful, but
 * WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * General Public License for more details.
 *
 * ============================================================ */

import java.util.UUID;

import java.io.Reader;
import java.io.FileReader;
import java.io.BufferedReader;
import java.io.File;
import java.io.IOException;

import rsb.EventId;
import rsb.ParticipantId;

public class event_id {

    public static void main(String[] args) throws IOException{
	File file = new File("data/event-id-cases.txt");
	BufferedReader reader = new BufferedReader(new FileReader(file));
	while (reader.ready()) {
	    String line = reader.readLine();
	    String[] tokens = line.split(" ");
	    ParticipantId originId = new ParticipantId(tokens[0]);
	    long sequenceNumber = Long.parseLong(tokens[1], 16);
	    UUID expectedId = UUID.fromString(tokens[2]);
	    EventId id = new EventId(originId, sequenceNumber);
	    if (!expectedId.equals(id.getAsUUID())) {
		System.exit(1);
	    }
	}

    }

};

File: python/event_id.py
#!/usr/bin/env python
# ============================================================
#
# Copyright (C) 2011 Jan Moringen <jmoringe@techfak.uni-bielefeld.de>
#
# This program is free software; you can redistribute it
# and/or modify it under the terms of the GNU General
# Public License as published by the Free Software Foundation;
# either version 2, or (at your option)
# any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# ============================================================

import logging
import uuid

import rsb

if __name__ == "__main__":
    logging.basicConfig(level=logging.DEBUG,
                        format='%(asctime)s %(name)-12s %(levelname)-8s\n%(message)s')
    with open('data/event-id-cases.txt') as input:
        for line in input.readlines():
            origin, seqnum, expected = map(str.strip, line.split(' '))
            originId, seqnum, expectedId = \
                (uuid.UUID(hex = origin), int(seqnum, 16), uuid.UUID(hex = expected))
            event = rsb.Event(id = rsb.EventId(participantId = originId, sequenceNumber = seqnum))
            assert event.id.getAsUUID() == expectedId
